src/Tokenizer/Tokens.php

```diff
@@ -617,10 +617,13 @@ public function getCodeHash(): string
     public function getCollectionHash(): string
     {
         if (null === $this->collectionHash) {
-            $this->collectionHash = md5(
-                $this->getCodeHash()
+            $countThis = \count($this);
+
+            $this->collectionHash = hash(
+                'xxh128',
```

Comment:
we shall be OK with XXH3

Reply:
https://stackoverflow.com/questions/3665247/fastest-hash-for-non-cryptographic-uses

We should use "xxh128" otherwise with many files birthday paradox can easily kick in.

Reply:
xxh3 is only 64 bit long, ie. the birthday paradox will apply since 4G files, which sounds a lot, but in reality a collision is not unlikely for someone.

Thus xxh128 should be preffered over xxh3.

👍 for this PR. I also belive different hash algo for different php runtine is ok.